### PR TITLE
monthly pull request 06.12.18

### PR DESCRIPTION
**Pull request by GerritRiesch94**

### CODE REVIEW

File: nbproject/Makefile-Release.mk [hunk 1]
```diff
@@ -88,17 +91,27 @@ LDLIBSOPTIONS=`pkg-config --libs lua-5.2`
 
 # Build Targets
 .build-conf: ${BUILD_SUBPROJECTS}
-	"${MAKE}"  -f nbproject/Makefile-${CND_CONF}.mk ${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/amos-ss17-proj4
+	"${MAKE}"  -f nbproject/Makefile-${CND_CONF}.mk ${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/car-simulator
```
Comment (tjaehnel) [inline nbproject/Makefile-Release.mk]: Thank you for that change!

File: resources/example_doip.lua [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Main = {
+	ANNOUNCE_NUM = 3,
+	ANNOUNCE_INT = 500,			-- in milliseconds
```
Comment (tjaehnel) [inline resources/example_doip.lua]: Abbreviation INT could be hard to understand.
Use full word instead ANNOUNCE_INTERVAL.

File: src/doip_lua_script.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
+#include "doip_lua_script.h"
+
+/**
+ * Default Constructor if no lua file was found.
+ */
+DoipLuaScript::DoipLuaScript(){ //if there is no config for the doip server or the ecus this default configuration will be used
```
Comment (tjaehnel) [inline src/doip_lua_script.cpp]: Not sure if "DoIPLuaScript" is a good name.
Maybe "DoIPConfigurationFile" is better

File: src/doip_lua_script.cpp [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    
+    std::cout<<"Setting Default Configuration for the DoIPServer"<<std::endl;
+    A_DoIP_Announce_Num = 3;
+    A_DoIP_Announce_Interval = 500;
+    vin_ = "00000000000000000";
```
Comment (tjaehnel) [inline src/doip_lua_script.cpp]: What does the trailing underscore mean?

File: src/doip_simulator.cpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+ */
+DoIPSimulator::DoIPSimulator() {
+    doipserver = new DoIPServer();
+}
+
+/**
+ * Initialize server instance with required callbacks, start the doip server
+ */
+void DoIPSimulator::start() {
+    DiagnosticCallback cb = std::bind(&DoIPSimulator::receiveFromLib, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
```
Comment (tjaehnel) [inline src/doip_simulator.cpp]: Is there a possibility to make callback registration a bit nicer?

File: src/doip_simulator.cpp [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        return false;
+    }
+  
+    //positiv ack
+    ackType = PayloadType::DIAGNOSTICPOSITIVEACK;
+    ackCode = 0x00;
+    
+    //negative ack
+    //ackType = PayloadType::DIAGNOSTICNEGATIVEACK;
+    //ackCode = 0x02;
```
Comment (Isafjoerdur) [inline src/doip_simulator.cpp]: Why is this part with the negative acknowledgement commented out? Is this intended/wanted?

File: src/start_arguments.cpp [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+//Initialize extern variables
+bool startargs::doip_flag = false;
+bool startargs::can_flag = false;
+std::string startargs::can_device;
+
+/**
+ * Parses the arguments when starting the car simulator.
+ * Usage:   ./car-simulator -d vcan0    for doip and can simulation
+ *          ./car-simulator             for doip and can simulation
+ *          ./car-simualtor -d          for doip simulation
```
Comment (Isafjoerdur) [inline src/start_arguments.cpp]: A small spelling mistake occurs at the above line for the arguments to start the car simulator. Should be "./car-simulator -d".

File: src/doip_simulator.cpp (hunk 5 above)
Comment (sascha15432) [inline src/doip_simulator.cpp]: Variable name "cb" is not  self-descriptive

File: src/doip_lua_script.cpp (hunk 4 above)
Comment (GerritRiesch94) [reply src/doip_lua_script.cpp]: nothing specific, it just should seperate the namings of some variables used in this function. I will change that

File: resources/example_doip.lua (hunk 2 above)
Comment (GerritRiesch94) [reply resources/example_doip.lua]: You are right, that is not a meaningful naming

File: src/doip_lua_script.cpp (hunk 3 above)
Comment (GerritRiesch94) [reply src/doip_lua_script.cpp]: We took the naming from the "ecu_lua_script" file that creates the simulated ecus from the script files and so does the "doip_lua_script" file for the DoIPServer instance but we can change that too

File: src/start_arguments.cpp (hunk 7 above)
Comment (GerritRiesch94) [reply src/start_arguments.cpp]: thank you, we will change that

File: src/doip_lua_script.cpp (hunk 3 above)
Comment (tjaehnel) [reply src/doip_lua_script.cpp]: I think that the context of these things are different. The ecu_lua_script is a _simulation_ file for an ECU. I agree that the name is not perfect, but at least it is a _script_ that contains executable parts.
On the other hand the doip_lua_script is a more or less _static configuration_ of the DoIPServer and not meant to be executed like a script. So I would somehow remove the word _script_ from the class name.

File: src/doip_simulator.cpp [hunk 8]
```diff
@@ -14,9 +14,26 @@ void DoIPSimulator::start() {
     DiagnosticCallback cb = std::bind(&DoIPSimulator::receiveFromLib, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
     DiagnosticMessageNotification dmn = std::bind(&DoIPSimulator::diagMessageReceived, this, std::placeholders::_1);
     doipserver->setCallback(cb, dmn);
+    //doipserver->setEIDdefault();    //setting EID to Default Value -> Address of the first ethernet controller
```
[the remark this answers is not in the document]
Comment (AndiAn94) [reply src/doip_simulator.cpp]: This was already changed. Anyway, thank you for the hint.

File: src/doip_simulator.cpp (hunk 5 above)
Comment (AndiAn94) [reply src/doip_simulator.cpp]: Thank you, changed this and some other variable names to hopefully more self-descriptive names.

File: src/doip_simulator.cpp (hunk 6 above)
Comment (AndiAn94) [reply src/doip_simulator.cpp]: This was inserted to test the method but we never fully removed it. Thank you for the hint, we removed some commented code.

**Comment by tjaehnel** [comment]
I've finally integrated the DoIP library into the CarSimulator.
Since a lot changed on CarSimulator in the meantime and also on the DoIP Library, it was not possible to just merge this PR.
But I've taken over most of the code from this PR: https://github.com/AVL-DiTEST-DiagDev/car-simulator/pull/15

**Pull request closed without merging**